### PR TITLE
Add server ACK for consumable

### PR DESCRIPTION
**Pull request by gerard-morera**
### :pushpin: References
* **Issue:** Fixes Candy Check [Issue](https://github.com/jnbt/candy_check/issues/48)

### :dart: What is the goal?
Perform server purchases acknowledgement.

Purchases ACK is used to confirm to google play that the purchase flow happened as expected. If the ACK does not happen the purchase is refunded within 4 minutes.

If the ACK is done from the client-side and the validation to the server fails it will make users that have paid not having it's correspondent capabilities active. This [candy_check](https://github.com/jnbt/candy_check) new feature will let acknowledge the purchase after capabilities grant.

### :memo: How is it being implemented?
Following Candy Check patterns and calling to Google API ACK service.

### CODE REVIEW

File: lib/candy_check/play_store/product_acknowledgements/response.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  module PlayStore
+    module ProductAcknowledgements
+      class Response
+        def initialize(result:, error_data:)
+          @result = result
+          @error_data = error_data
+        end
+
+        def acknowleged?
+          !!result
```
Comment (gerard-morera) [inline lib/candy_check/play_store/product_acknowledgements/response.rb:11]: FYI `result` when comes is an empty string.

File: lib/candy_check/play_store/product_acknowledgements/acknowledgement.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          acknowlege!
+
+          CandyCheck::PlayStore::ProductAcknowledgements::Response.new(
+            result: @response[:result], error_data: @response[:error_data])
+        end
+
+        private
+
+        def acknowlege!
+          service = CandyCheck::PlayStore::AndroidPublisherService.new
```
Comment (oricodes89) [inline lib/candy_check/play_store/product_acknowledgements/acknowledgement.rb:35]: I think this class shouldn't have the remote call. Looking at the architecture of the gem, this simply encapsulates the response and works as an entity.

File: lib/candy_check/play_store/acknowledger.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      end
+
+      def acknowledge_product_purchase(package_name:, product_id:, token:)
+        acknowledger = CandyCheck::PlayStore::ProductAcknowledgements::Acknowledgement.new(
+          package_name: package_name,
+          product_id: product_id,
+          token: token,
+          authorization: @authorization,
+        )
+        acknowledger.call!
```
Comment (oricodes89) [inline lib/candy_check/play_store/acknowledger.rb:15]: Like I say [here](https://github.com/Badiapp/candy_check/pull/1/files#r444000801), it is weird that the `Acknowledger` does not have the logic to acknowledge things itself.

This class does nothing basically 🤔

File: lib/candy_check/play_store/product_acknowledgements/acknowledgement.rb (hunk 2 above)
Comment (oricodes89) [reply lib/candy_check/play_store/product_acknowledgements/acknowledgement.rb:35]: Perhaps it [is the other class](https://github.com/Badiapp/candy_check/pull/1/files#r444002583) that can be skipped if you want to add everything here?

File: lib/candy_check/play_store/acknowledger.rb (hunk 3 above)
Comment (gerard-morera) [reply lib/candy_check/play_store/acknowledger.rb:15]: I agree that is a wrapper of `acknowledgement` IMO it works as [Verifier](https://github.com/Badiapp/candy_check/blob/feature/add-server-ack-for-consumable/lib/candy_check/play_store/verifier.rb#L5) and [ProductVerification](https://github.com/Badiapp/candy_check/blob/feature/add-server-ack-for-consumable/lib/candy_check/play_store/product_purchases/product_verification.rb#L6). I think the reason is to have a simpler public API.

Comment (oricodes89) [reply lib/candy_check/play_store/acknowledger.rb:15]: Just saw that this "dumb" class is how the gem operates with the ["verifier"](https://github.com/Badiapp/candy_check/blob/master/lib/candy_check/play_store/verifier.rb)

**Review by oricodes89 — APPROVED** [review]
LGTM, the gem handles responses in a confusing way.

File: lib/candy_check/play_store/acknowledger.rb (hunk 3 above)
Comment (oricodes89) [reply lib/candy_check/play_store/acknowledger.rb:15]: Go ahead!

**Pull request merged**